### PR TITLE
Added default configs for commands

### PR DESCRIPTION
**Pull request by PlasmaPower**
Needs testing

### CODE REVIEW

**Comment by amikhalev** [comment]
I'm not sure if I agree with the Default postfix to denote defaults. What do you think about having a command with a name of Defaults or something containing the default config?

File: src/Common/Config/Kremlin.h [hunk 1]
```diff
@@ -41,6 +41,8 @@ class Kremlin {
 	};
 	/// Map of strings to CommandDetails
 	static std::map<std::string, CommandDetails> commands;
+	/// Map of command names to default configs
+	static std::map<std::string, YAML::Node> defaultConfigs;
```
Comment (amikhalev) [inline src/Common/Config/Kremlin.h]: Couldn't this be stored in the CommandDetails struct?

File: src/Common/Config/Kremlin.cpp [hunk 2]
```diff
@@ -152,6 +163,19 @@ bool Kremlin::CreateGenericCommandGroups(bool lastTry) {
...
+					defaultConfig = YAML::Node();
+				} else {
+					continue;
+				}
+			} else {
+				defaultConfig = defaultConfigs[fullName];
+			}
+			for (YAML::Node subnode : defaultConfig) {
+				commandConfig.push_back(subnode);
+			}
```
Comment (amikhalev) [inline src/Common/Config/Kremlin.cpp]: I think this is only executed for CommandGroup, and those generally don't need defaults. I think Zach and I were thinking more of something for regular commands.

Comment (PlasmaPower) [inline src/Common/Config/Kremlin.cpp]: Oops yeah. I put that in the wrong part of the code.

**Comment by PlasmaPower** [comment]
I think having a postfix would allow us to organize the config more. Remember when we had a `$` object? Also, I'm redoing this because I accidentally did this for command groups.

**Comment by PlasmaPower** [comment]
How about we add "subCommands" which are basically versions of a command with modified configurations? Another option is to have a `- name: Default` node, but that would offer less extensibility.

File: src/Common/Config/Kremlin.cpp (hunk 2 above)
Comment (amikhalev) [inline src/Common/Config/Kremlin.cpp]: Also, I believe that the way this is implemented, it only works for sequences and crashes if the command config isn't a sequence, and also the default config would override the command config which isn't what we want.

Comment (PlasmaPower) [inline src/Common/Config/Kremlin.cpp]: Commands must be a sequence as they must have a `- name: `.

**Pull request closed without merging**